src/uniformscalingmap.jl

```diff
@@ -19,20 +19,48 @@ LinearAlgebra.transpose(A::UniformScalingMap) = A
 LinearAlgebra.adjoint(A::UniformScalingMap)   = UniformScalingMap(conj(A.λ), size(A))
 
 # multiplication with vector
+Base.:(*)(A::UniformScalingMap, x::AbstractVector) = A.λ * x
```

Comment:
Unlike the mutating version, this does not check whether the vector length is compatible (not your fault, was so in the original version). But maybe such a check can be added for consistency?